lib/winapi/winnt.h

```diff
@@ -13,4 +14,22 @@ typedef LPCWSTR LPCTSTR;
 typedef LPCSTR LPCTSTR;
 #endif
 
+static LONG64 InterlockedExchange64(LONG64 volatile *Target, LONG64 Value)
+{
+    return __atomic_exchange_n(Target, Value, __ATOMIC_ACQ_REL);
+}
+
+static PVOID InterlockedExchangePointer(PVOID volatile *Target, PVOID Value)
+{
+    return __atomic_exchange_n(Target, Value, __ATOMIC_ACQ_REL);
+}
+
+static PVOID InterlockedCompareExchangePointer(PVOID volatile *Destination, PVOID Exchange, PVOID Comperand)
+{
+  PVOID hack = *Destination; //FIXME: This isn't atomic
+  int result = __atomic_compare_exchange_n (Destination, Comperand, Exchange, 0, __ATOMIC_ACQ_REL, __ATOMIC_ACQUIRE);
+  return result ? Comperand : hack;
```

Comment:
Clang already has these: https://github.com/llvm-mirror/clang/blob/master/include/clang/Basic/Builtins.def#L817

So it seems that this merely needs a function or macro to be created with the right signature.